src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.macOS.cs

```diff
@@ -84,7 +80,13 @@ private void UpdateTextBoxView()
 				}
 				else
 				{
-					_textBoxView = new TextBoxView(this) { UsesSingleLineMode = AcceptsReturn || TextWrapping != TextWrapping.NoWrap };
+					var textWrapping = TextWrapping;
+					var usesSingleLineMode = !(AcceptsReturn || textWrapping != TextWrapping.NoWrap);
+					_textBoxView = new TextBoxView(this)
+					{
+						UsesSingleLineMode = usesSingleLineMode,
+						LineBreakMode = textWrapping == TextWrapping.WrapWholeWords ? NSLineBreakMode.ByWordWrapping : NSLineBreakMode.CharWrapping,
```

Comment:
it's kinda odd-looking that you break out `usesSingleLineMode` into a variable but calculate `LineBreakMode` inline

Reply:
That's because the reading of it was not obvious and was probably the cause of the bug. But there's not such ambiguity when I added the LineBreakMode feature.